pkg/imgpkg/cmd/copy_repo_src.go

```diff
@@ -113,70 +125,72 @@ func (c CopyRepoSrc) getSourceImages() (*ctlimgset.UnprocessedImageRefs, error)
 				Tag:       bundleLock.Bundle.Tag,
 			})
 
-			return unprocessedImageRefs, nil
+			return unprocessedImageRefs, bundles, nil
 
 		case imagesLock != nil:
+			c.logger.Tracef("get images from ImagesLock file\n")
 			for _, img := range imagesLock.Images {
 				plainImg := plainimage.NewPlainImage(img.Image, c.registry)
 
 				ok, err := ctlbundle.NewBundleFromPlainImage(plainImg, c.registry).IsBundle()
 				if err != nil {
-					return nil, err
+					return nil, nil, err
 				}
 				if ok {
-					return nil, fmt.Errorf("Unable to copy bundles using an Images Lock file (hint: Create a bundle with these images)")
+					return nil, nil, fmt.Errorf("Unable to copy bundles using an Images Lock file (hint: Create a bundle with these images)")
 				}
 
 				unprocessedImageRefs.Add(ctlimgset.UnprocessedImageRef{DigestRef: plainImg.DigestRef()})
 			}
-			return unprocessedImageRefs, nil
+			return unprocessedImageRefs, nil, nil
 
 		default:
 			panic("Unreachable")
 		}
 
 	case c.ImageFlags.Image != "":
+		c.logger.Tracef("copy single image\n")
 		plainImg := plainimage.NewPlainImage(c.ImageFlags.Image, c.registry)
 
 		ok, err := ctlbundle.NewBundleFromPlainImage(plainImg, c.registry).IsBundle()
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
 		if ok {
-			return nil, fmt.Errorf("Expected bundle flag when copying a bundle (hint: Use -b instead of -i for bundles)")
+			return nil, nil, fmt.Errorf("Expected bundle flag when copying a bundle (hint: Use -b instead of -i for bundles)")
 		}
 
 		unprocessedImageRefs.Add(ctlimgset.UnprocessedImageRef{DigestRef: plainImg.DigestRef(), Tag: plainImg.Tag()})
-		return unprocessedImageRefs, nil
+		return unprocessedImageRefs, nil, nil
 
 	default:
-		bundle, imageRefs, err := c.getBundleImageRefs(c.BundleFlags.Bundle)
+		c.logger.Tracef("copy bundle\n")
+		bundles, imagesRef, err := c.getBundleImageRefs(c.BundleFlags.Bundle)
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
 
-		for _, img := range imageRefs {
+		for _, img := range imagesRef {
 			unprocessedImageRefs.Add(ctlimgset.UnprocessedImageRef{DigestRef: img.PrimaryLocation()})
 		}
 
-		unprocessedImageRefs.Add(ctlimgset.UnprocessedImageRef{DigestRef: bundle.DigestRef(), Tag: bundle.Tag()})
+		unprocessedImageRefs.Add(ctlimgset.UnprocessedImageRef{DigestRef: c.originBundle.DigestRef(), Tag: c.originBundle.Tag()})
 
-		return unprocessedImageRefs, nil
+		return unprocessedImageRefs, bundles, nil
 	}
 }
 
-func (c CopyRepoSrc) getBundleImageRefs(bundleRef string) (*ctlbundle.Bundle, []lockconfig.ImageRef, error) {
-	bundle := ctlbundle.NewBundle(bundleRef, c.registry)
-
-	imgLock, err := bundle.AllImagesLock(c.Concurrency)
+func (c *CopyRepoSrc) getBundleImageRefs(bundleRef string) ([]*ctlbundle.Bundle, ctlbundle.ImagesRef, error) {
+	c.originBundle = ctlbundle.NewBundle(bundleRef, c.registry)
```

Comment:
instead of saving this off, we should just return rootBundle. looks like `...) (*Bundle, []*ctlbundle.Bundle, ctlbundle.ImagesRef, error)`. otherwise it's just a hidden contract.

Reply:
fixed